android/modules/ui/src/java/ti/modules/titanium/ui/widget/tableview/TiBaseTableViewItem.java

```diff
@@ -48,67 +51,74 @@ public TiBaseTableViewItem(Activity activity)
+						break;
 				}
-				if (checkIndicatorBitmap == null) {
-					String path =
-						"/org/appcelerator/titanium/res/drawable/btn_check_buttonless_on_48.png"; // default medium
-					switch (density) {
-						case DisplayMetrics.DENSITY_HIGH:
-							path = "/org/appcelerator/titanium/res/drawable/btn_check_buttonless_on_72.png";
-							break;
-						case DisplayMetrics.DENSITY_MEDIUM:
-							path = "/org/appcelerator/titanium/res/drawable/btn_check_buttonless_on_48.png";
-							break;
-						case DisplayMetrics.DENSITY_LOW:
-							path = "/org/appcelerator/titanium/res/drawable/btn_check_buttonless_on_36.png";
-							break;
-					}
-
-					if (Build.VERSION.SDK_INT >= 9 && density == DisplayMetrics.DENSITY_XHIGH) {
-						path = "/org/appcelerator/titanium/res/drawable/btn_check_buttonless_on_96.png";
-					}
-
-					if (Build.VERSION.SDK_INT >= 16 && density >= DisplayMetrics.DENSITY_XXHIGH) {
-						path = "/org/appcelerator/titanium/res/drawable/btn_check_buttonless_on_144.png";
-					}
-
-					if (Build.VERSION.SDK_INT >= 18 && density >= DisplayMetrics.DENSITY_XXXHIGH) {
-						path = "/org/appcelerator/titanium/res/drawable/btn_check_buttonless_on_192.png";
-					}
-
-					checkIndicatorBitmap = BitmapFactory.decodeStream(KrollDict.class.getResourceAsStream(path));
+
+				if (Build.VERSION.SDK_INT >= 9 && density == DisplayMetrics.DENSITY_XHIGH) {
+					res = "btn_more_96";
+				}
+
+				if (Build.VERSION.SDK_INT >= 16 && density >= DisplayMetrics.DENSITY_XXHIGH) {
+					res = "btn_more_144";
+				}
+
+				if (Build.VERSION.SDK_INT >= 16 && density >= DisplayMetrics.DENSITY_XXXHIGH) {
+					res = "btn_more_192";
+				}
+
+				try {
+					childIndicatorBitmap =
+						BitmapFactory.decodeStream(resources.openRawResource(TiRHelper.getResource("drawable." + res)));
+				} catch (Exception e) {
+					Log.w(TAG, "Drawable not found for resource id: " + res);
+				}
+			}
+			if (checkIndicatorBitmap == null) {
+				String res = "btn_check_buttonless_on_48"; // default medium
+				switch (density) {
+					case DisplayMetrics.DENSITY_HIGH:
+						res = "btn_check_buttonless_on_72";
+						break;
+					case DisplayMetrics.DENSITY_MEDIUM:
+						res = "btn_check_buttonless_on_48";
+						break;
+					case DisplayMetrics.DENSITY_LOW:
+						res = "btn_check_buttonless_on_36";
+						break;
+				}
+
+				if (Build.VERSION.SDK_INT >= 9 && density == DisplayMetrics.DENSITY_XHIGH) {
+					res = "btn_check_buttonless_on_96";
+				}
+
+				if (Build.VERSION.SDK_INT >= 16 && density >= DisplayMetrics.DENSITY_XXHIGH) {
+					res = "btn_check_buttonless_on_144";
+				}
+
+				if (Build.VERSION.SDK_INT >= 18 && density >= DisplayMetrics.DENSITY_XXXHIGH) {
+					res = "btn_check_buttonless_on_192";
+				}
+
+				try {
+					checkIndicatorBitmap =
+						BitmapFactory.decodeStream(resources.openRawResource(TiRHelper.getResource("drawable." + res)));
+				} catch (Exception e) {
+					Log.w(TAG, "Drawable not found for resource id: " + res);
```

Comment:
Hmm... the old code was loading density scaled images the hard way. Not great.

I noticed you moved the images  to "res/drawable". This is definitely the better move. But we should prefix the image files with "ti_" to avoid resource name collision with the app dev's own resource files and Android libraries.

We can also get rid of our density handling code if we move the following images to the appropriate res directory:
- "btn_check_buttonless_on_36" -> "res/drawable/ldpi/ti_btn_check_buttonless"
- "btn_check_buttonless_on_48" -> "res/drawable/mdpi/ti_btn_check_buttonless"
- "btn_check_buttonless_on_72" -> "res/drawable/hdpi/ti_btn_check_buttonless"
- "btn_check_buttonless_on_96" -> "res/drawable/xhdpi/ti_btn_check_buttonless"
- "btn_check_buttonless_on_144" -> "res/drawable/xxhdpi/ti_btn_check_buttonless"
- "btn_check_buttonless_on_192" -> "res/drawable/xxxhdpi/ti_btn_check_buttonless"

I think our kitchensink-v2 sample app displays the above images.

Reply:
We also don't need to use the `BitmapFactory` anymore since this class has methods `createHasChildDrawable()` and `createHasCheckDrawable()` which wraps the 2 loaded bitmaps with drawables. We should call the `Resources,getDrawable()` method instead and change member variables `childIndicatorBitmap` and `checkIndicatorBitmap` to type `Drawable`.

Reply:
I have cleaned up the existing code but `TiRHelper.getImageRessource()` should really be refactored to use DPI paths.